rmw_fastrtps_shared_cpp/src/reader_info.hpp

```diff
@@ -97,19 +99,13 @@ class ReaderInfo : public eprosima::fastrtps::rtps::ReaderListener
     mapmutex.unlock();
 
     if (trigger) {
-      rmw_ret_t ret = rmw_trigger_guard_condition(graph_guard_condition_);
-      if (ret != RMW_RET_OK) {
-        RCUTILS_LOG_ERROR_NAMED(
-          "rmw_fastrtps_cpp",
-          "failed to trigger graph guard condition: %s",
-          rmw_get_error_string_safe())
-      }
+	  graph_guard_condition_->trigger();
```

Comment:
Where does the new code preserve the error message?

Reply:
The new code now keeps a pointer to an internal GuardCondition object, and calls its trigger method directly. The only possible failure would be an incorrect implementation_identifier and, as the guard condition is created internally when creating the node, the implementation_identifier is not going to be wrong anyway.